t/10-jobs.t

```diff
@@ -942,4 +949,50 @@ subtest 'special cases when restarting job via Minion task' => sub {
     $test->([99945], 'failed', 'some error', 'error if an error occurs and there are no attempts left');
 };
 
+subtest 'git log diff' => sub {
+    my $job_mock = Test::MockModule->new('OpenQA::Schema::Result::Jobs', no_auto => 1);
+    $job_mock->redefine(
+        run_cmd_with_log_return_error => sub ($cmd, %opt) {
+            my $rc = 0;
+            my $stdout = '';
+            if ("@$cmd" =~ m/rev-list --count/) {
+                if ("@$cmd" =~ m/revlistfail/) { $stdout = "git failed"; $rc = 1; }
+                elsif ("@$cmd" =~ m/nonumber/) { $stdout = "NaN"; }
+                else { $stdout = 10; }
+            }
+            elsif ("@$cmd" =~ m/diff --stat/) {
+                if ("@$cmd" =~ m/difffail/) { $stdout = "git failed"; $rc = 1; }
+                else { $stdout = "2 files changed"; }
+            }
+            return {stdout => $stdout, return_code => $rc, stderr => ''};
+        });
+    my %_settings = %settings;
+    $_settings{TEST} = 'L';
+    my $job = _job_create(\%_settings);
+
+    my $too_big = $job->git_diff('/foo', '123..456', 5);
+    like $too_big, qr{Too many commits}, 'Too many commits';
+
+    my $warning = warning {
+        my $non_numeric = $job->git_diff('/foo', 'nonumber..123456', 10);
+        like $non_numeric, qr{Cannot display diff because of a git problem};
```

Comment:
Can you please add test description strings in all cases

Reply:
done